From: Keith Packard <keithp@koto.keithp.com>
Date: Wed, 8 Aug 2007 19:16:12 +0000 (-0700)
Subject: RRScanOldConfig cannot use RRFirstOutput before output is configured.
X-Git-Url: http://gitweb.freedesktop.org/?p=xorg/xserver.git;a=commitdiff;h=b2dcfbca2441ca8c561f86a78a76ab59ecbb40e4

RRScanOldConfig cannot use RRFirstOutput before output is configured.

RRFirstOutput returns the first active output, which won't be set until
after RRScanOldConfig is finished running. Instead, just use the first
output (which is the only output present with an old driver, after all).
---

--- randr/rrinfo.c.orig	2007-07-31 18:37:36.000000000 -0400
+++ randr/rrinfo.c
@@ -105,7 +105,7 @@ RRScanOldConfig (ScreenPtr pScreen, Rota
 #endif
     }
 
-    output = RRFirstOutput (pScreen);
+    output = pScrPriv->outputs[0];
     if (!output)
 	return;
     crtc = output->crtc;
